types/src/transaction/authenticator.rs

```diff
@@ -1067,6 +1070,9 @@ pub enum AnyPublicKey {
     Keyless {
         public_key: KeylessPublicKey,
     },
+    FederatedKeyless {
+        public_key: FederatedKeylessPublicKey,
+    },
```

Comment:
so this modifies the on-chain `Transaction` structure directly... probably need to feature gate this?
and i wonder what's a good way. A correct way i know is to modify consensus to not wrap a `FederatedKeyless` transaction when the feature flag is off, but we didn't do it while adding the `Keyless` variant?

Reply:
+1 feature gate

Reply:
Need to feature gate, yes.

But the change to `AnyPublicKey` is not a big problem: we update enums all the time.

Reply:
Added feature gating!